client/packages/executor/src/prometheus.ts

```diff
@@ -0,0 +1,113 @@
+import http from "http";
+import { Registry, Counter, collectDefaultMetrics } from "prom-client";
+
+export class Prometheus {
+  register: Registry;
+
+  // State
+  isActive: boolean;
+
+  // Metrics
+  // Number of bids
+  bids: Counter;
+
+  // Number of events
+  events: Counter;
+
+  // Number of times circuit rpc server has disconnected
+  circuitDisconnects: Counter;
+
+  // Information on circuit disconnections
+  circuitDisconnected: Counter;
+
+  // Number of times no bid and no competition
+  noBidAndNoCompetition: Counter;
+
+  // Number of times no bid but competition
+  noBidButCompetition: Counter;
+
+  // Number of times been out bid
+  beenOutBid: Counter;
+
+  constructor() {
+    this.register = new Registry();
+    this.createMetrics();
+  }
+
+  // Creates all the metrics for Prometheus
+  createMetrics() {
+    // Collects default metrics
+    collectDefaultMetrics({ register: this.register });
+    this.bids = new Counter({
+      name: "bids",
+      help: "Number of bids",
+      registers: [this.register],
+    });
+
+    this.events = new Counter({
+      name: "events",
+      help: "Number of events",
+      registers: [this.register],
+    });
+
+    this.circuitDisconnected = new Counter({
+      name: "circuit_disconnect",
+      help: "Information on circuit disconnections",
+      registers: [this.register],
+      labelNames: ["endpoint"],
+    });
+
+    this.circuitDisconnects = new Counter({
+      name: "circuit_disconnects_total",
+      help: "Number of times circuit rpc server has disconnected",
+      registers: [this.register],
+    });
+
+    this.noBidAndNoCompetition = new Counter({
+      name: "no_bid_and_no_competition",
+      help: "Number of times no bid and no competition",
+      registers: [this.register],
+    });
+
+    this.noBidButCompetition = new Counter({
+      name: "no_bid_but_competition",
+      help: "Number of times no bid but competition",
+      registers: [this.register],
+    });
+
+    this.beenOutBid = new Counter({
+      name: "been_out_bid",
+      help: "Number of times been out bid",
+      registers: [this.register],
+    });
+
+    this.startServer();
+  }
+
+  startServer() {
+    const port = 8080;
```

Comment:
maybe not hardcode the port but it expose it through config

Reply:
okay

Reply:
Fixed, see commit [a87c786](https://github.com/t3rn/t3rn/pull/1051/commits/a87c78633419a638ae002349ee631964e61e102a).